packages/modules/pricing/src/repositories/pricing.ts

```diff
@@ -163,6 +163,7 @@ export class PricingRepository
         pl_rules_count: "price.pl_rules_count",
         price_list_type: "price.pl_type",
         price_list_id: "price.price_list_id",
+        all_rules_count: knex.raw("price.rules_count + price.pl_rules_count"),
```

Comment:
don't we need a COALESCE here? "pl_rules_count" could be NULL no?

Reply:
Both rules count are non-nullable and default to 0, so we should be good

Reply:
At a first glance `price.pl_rules_count` apparently comes from `pl_rules_count: "pl.rules_count"`, and that is a left join with price_list.
that why I asked if that can be null.

Reply:
it seams that the pl_price_count is indeed retrieved through a left join and it means by default that the right side is optional and therefore if you dont have a matching pl tuple the value will not be present which is why carlos is suggesting to use a coalescing (either in the original sub query or before using it, i would suggest to coalesce in the original select to be always defaulted) as it can be null. The only time it cant be null is when the left join as a match and therefore the default value at least will take place or if you were in the case of an inner join. 

Feel free to let us know if we are wrong but could you run a little test query (just one select with one left join) where you select a price that does have a matching price list and let us know what the select returns you, i believe the count will be null 🙏

Maybe we are missing some context and in that case feel free to complement.

Reply:
Ah yes indeed, I actually missed the fact that it can be null, nice catch guys 👍